x-pack/plugins/endpoint/public/applications/endpoint/store/endpoint_list/saga.ts

```diff
@@ -6,17 +6,29 @@
 
 import { CoreStart } from 'kibana/public';
 import { SagaContext } from '../../lib';
-import { EndpointListAction } from './action';
+import { AppAction } from '../action';
+import { endpointListPageIndex, endpointListPageSize } from './selectors';
 
 export const endpointListSaga = async (
-  { actionsAndState, dispatch }: SagaContext<EndpointListAction>,
+  { actionsAndState, dispatch }: SagaContext<AppAction>,
   coreStart: CoreStart
 ) => {
   const { post: httpPost } = coreStart.http;
 
-  for await (const { action } of actionsAndState()) {
-    if (action.type === 'userEnteredEndpointListPage') {
-      const response = await httpPost('/api/endpoint/endpoints');
+  for await (const { action, state } of actionsAndState()) {
+    if (
+      (action.type === 'userNavigatedToPage' && action.payload === 'managementPage') ||
+      action.type === 'userPaginatedEndpointListTable'
+    ) {
+      const pageIndex = endpointListPageIndex(state.endpointList);
+      const pageSize = endpointListPageSize(state.endpointList);
+      const response = await httpPost('/api/endpoint/endpoints', {
+        body: JSON.stringify({
+          paging_properties: [{ page_index: pageIndex }, { page_size: pageSize }],
+        }),
+      });
+      // temp: request_page_index to reflect user request page index, not es page index
+      response.request_page_index = pageIndex;
       dispatch({
         type: 'serverReturnedEndpointList',
         payload: response,
```

Comment:
@oatkiller HALP

I heard you were a TS Master Wizard - why are we NOT getting an TS error here? `response` is `any` (because we did not tell `post` what the response from the API looks like. I should have thrown an error when trying to assign `any` to `payload` which has an explicit type on it.

WHYYYYYYY 😄

Reply:
you can assign `any` to anything.